lib/uklibparam/include/uk/libparam.h

```diff
@@ -187,18 +130,16 @@ UK_CTOR_PRIO(UK_LIBPARAM_LIBDESC_CTOR, 0);
 
 #define __UK_LIBPARAM_PARAM_DEFINE(arg_var, arg_addr, arg_type, arg_count, \
 				   arg_desc)				\
-	static struct uk_libparam_param __UK_LIBPARAM_PARAM_NAME(arg_var) = { \
-		.name  = STRINGIFY(arg_var),				\
-		.desc  = arg_desc,					\
-		.type  = _UK_LIBPARAM_PARAM_TYPE_##arg_type,		\
-		.count = arg_count,					\
-		.addr  = arg_addr					\
+	static const struct uk_libparam_param				\
+	__used __section("." STRINGIFY(UK_LIBPARAM_PARAMSECTION_NAME))	\
+	__align(8) __UK_LIBPARAM_PARAM_NAME(arg_var) = {		\
+		.prefix = STRINGIFY(UK_LIBPARAM_LIBPREFIX),		\
+		.name   = STRINGIFY(arg_var),				\
+		.desc   = arg_desc,					\
+		.type   = _UK_LIBPARAM_PARAM_TYPE_##arg_type,		\
+		.count  = arg_count,					\
+		.addr   = arg_addr					\
```

Comment:
I followed the existing convention, see the diff. Generally speaking I'm okay with either way, but unless I missed something I didn't find anything related in https://unikraft.org/docs/contributing/coding-conventions. Until we have such a convention I would consider this a matter of personal style and stick to each library's convention.

```diff
-       static struct uk_libparam_param __UK_LIBPARAM_PARAM_NAME(arg_var) = { \
-               .name  = STRINGIFY(arg_var),                            \
-               .desc  = arg_desc,                                      \
-               .type  = _UK_LIBPARAM_PARAM_TYPE_##arg_type,            \
-               .count = arg_count,                                     \
-               .addr  = arg_addr                                       \
+       static const struct uk_libparam_param                           \
+       __used __section("." STRINGIFY(UK_LIBPARAM_PARAMSECTION_NAME))  \
+       __align(8) __UK_LIBPARAM_PARAM_NAME(arg_var) = {                \
+               .prefix = STRINGIFY(UK_LIBPARAM_LIBPREFIX),             \
+               .name   = STRINGIFY(arg_var),                           \
+               .desc   = arg_desc,                                     \
+               .type   = _UK_LIBPARAM_PARAM_TYPE_##arg_type,           \
+               .count  = arg_count,                                    \
+               .addr   = arg_addr                                      \
```

Reply:
That comma is a common pattern I've seen in most open source C-based projects. There are a few benefits such as easily being able to reorder lines later or easily add new statically initialized fields. I am not going to insist on a trailing comma, but I wanted to take a moment to explain why I am thinking that it is also objectively better, since I do not see any benefit to not adding it.

Reply:
Yes, as I said I do not object to that, in fact I agree. But I would like that we either formalize such things,  or refrain from ad-hoc reviewer's requests on unspecified things that are often a matter of personal taste (e.g. xmas tree). Saying that, if you feel it's important enough to open a PR for `coding-conventions` on the above, I'll vouch for this change.